code/core/src/shared/open-service/README.md

```
@@ -358,6 +375,119 @@ flowchart TD
   J --> K[writeOpenServiceStaticFiles outputDir]
 ```
 
+## Client Architecture (Multi-Master)
+
+Browser processes (manager and preview) each run their own full `ServiceRuntime` — identical in shape to the server-side one. State is reconciled peer-to-peer through Storybook's existing manager↔preview channel using a welcome-handshake + patch-broadcast protocol.
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)